src/test/java/io/appium/java_client/service/local/ServerBuilderTest.java

```diff
@@ -312,4 +318,47 @@ public void checkAbilityToStartServiceWithLogFileUsingShortFlag() {
         service.start();
         assertTrue(testLogFile.exists());
     }
+
+    @Test
+    public void checkAbilityToStartServiceUsingValidBasePathWithMultiplePathParams() {
+        String baseUrl = String.format("http://%s:%d/", BROADCAST_IP_ADDRESS, DEFAULT_APPIUM_PORT);
+        String basePath = "wd/hub";
+        service = new AppiumServiceBuilder().withArgument(BASEPATH, basePath).build();
+        service.start();
+        assertTrue(service.isRunning());
+        assertEquals(baseUrl + basePath + "/", service.getUrl().toString());
+    }
+
+    @Test
+    public void checkAbilityToStartServiceUsingValidBasePathWithSinglePathParams() {
+        String baseUrl = String.format("http://%s:%d/", BROADCAST_IP_ADDRESS, DEFAULT_APPIUM_PORT);
+        String basePath = "/wd/";
+        service = new AppiumServiceBuilder().withArgument(BASEPATH, basePath).build();
+        service.start();
+        assertTrue(service.isRunning());
+        assertEquals(baseUrl + basePath.substring(1), service.getUrl().toString());
+    }
+
+    @Test
+    public void checkAbilityToValidateBasePathForEmptyBasePath() {
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            service = new AppiumServiceBuilder().withArgument(BASEPATH, "").build();
+        });
+        assertEquals(INVALID_BASE_PATH_ERROR_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void checkAbilityToValidateBasePathForBlankBasePath() {
+        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+            service = new AppiumServiceBuilder().withArgument(BASEPATH, "   ").build();
+        });
+        assertEquals(INVALID_BASE_PATH_ERROR_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void checkAbilityToValidateBasePathForNullBasePath() {
+        assertThrows(NullPointerException.class, () -> {
+            service = new AppiumServiceBuilder().withArgument(BASEPATH, null).build();
```

Comment:
no need to init `service` variable here

Reply:
Updated in the new commit, thanks for highlighting